lemming.go

```diff
@@ -55,8 +57,57 @@ type Device struct {
 	stopped chan struct{}
 }
 
+// Option are device startup options for lemming.
+type Option func(*opt)
+
+type opt struct {
+	// deviceConfigJSON is the contents of the JSON document (prior to unmarshal).
+	deviceConfigJSON []byte
+	// tlsCredentials contains TLS credentials that can be used for a device.
+	tlsCredentials credentials.TransportCredentials
+}
+
+// resolveOpts applies all the options and returns a struct containing the result.
+func resolveOpts(opts []Option) *opt {
+	o := &opt{}
+	for _, opt := range opts {
+		opt(o)
+	}
+	return o
+}
+
+// WithDeviceConfig sets the startup config of the device to c.
+// Today we do not allow the configuration to be changed in flight, but this
+// can be implemented in the future.
+//
+// This DeviceOption is intended for standalone device testing.
+func WithDeviceConfig(c []byte) Option {
+	return func(o *opt) {
+		o.deviceConfigJSON = c
+	}
+}
+
+// WithTLSCredsFromFile loads the credentials from the specified cert and key file
+// and returns them such that they can be used for the gNMI and gRIBI servers.
+func WithTLSCredsFromFile(certFile, keyFile string) (Option, error) {
+	t, err := credentials.NewServerTLSFromFile(certFile, keyFile)
+	if err != nil {
+		return nil, err
+	}
+	return func(o *opt) {
+		o.tlsCredentials = t
+	}, nil
+}
+
+// WithTLSCreds returns a wrapper of TransportCredentials into a DevOpt.
+func WithTLSCreds(c credentials.TransportCredentials) Option {
```

Comment:
nit: if insecure.Credentials are meant to be used with the option then, call it WithCreds() or WithTransportCreds

Reply:
Done.